tests/testthat/test-dbQuoteString.R

```diff
@@ -0,0 +1,10 @@
+context("dbQuoteString")
+
+test_that("quoting works", {
+  if (!mysqlHasDefault()) skip("Test database not available")
+
+  con <- dbConnect(MySQL(), dbname = "test")
+  expect_equal(dbQuoteString(con, "\\'"), SQL("'\\\\\\''"))
+  expect_equal(dbQuoteString(con, "'"), SQL("'\\''"))
+  on.exit(dbDisconnect(con))
```

Comment:
The `on.exit()` guards premature exit, e.g. if the testing code errors:

```suggestion
  on.exit(dbDisconnect(con))
  expect_equal(dbQuoteString(con, "\\'"), SQL("'\\\\\\''"))
  expect_equal(dbQuoteString(con, "'"), SQL("'\\''"))
```